src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs

```diff
@@ -77,7 +78,23 @@ private static async Task<HttpResponseMessage> SendWithNtAuthAsync(HttpRequestMe
 
                         string challengeData = challenge.ChallengeData;
 
-                        string spn = "HTTP/" + authUri.IdnHost;
+                        // Need to use FQDN normalized host so that CNAME's are traversed.
+                        string spn;
+                        if (authUri.HostNameType == UriHostNameType.IPv6 || authUri.HostNameType == UriHostNameType.IPv4)
+                        {
+                            spn = "HTTP/" + authUri.IdnHost;
+                        }
+                        else
+                        {
+                            IPHostEntry result = await Dns.GetHostEntryAsync(authUri.IdnHost);
+                            spn = "HTTP/" + result.HostName;
+                        }
```

Comment:
Nit: maybe instead of doing:
```C#
string spn;
if (...)
{
    spn = "HTTP/" + authUri.IdnHost;
}
else
{
    ...
    spn = "HTTP/" + result.HostName;
}
```
do:
```C#
string spn;
if (...)
{
    spn = authUri.IdnHost;
}
else
{
    ...
    spn = result.HostName;
}
spn = "HTTP/" + spn;
```
?

Reply:
Why is this better?  Is seems slower because we are doing two assignments of `spn` field.  

I don't think it is really less lines of IL or memory because the "HTTP/" string should only occur once.

Reply:
> Why is this better? Is seems slower because we are doing two assignments of spn field.

From my perspective it's better because it's centralizing some logic, namely the concatenation of the "HTTP/" prefix.  There's also no additional field assignment, as `spn` here is a local, not a field.  And the IL is a bit longer otherwise, as there are two calls to Concat instead of one:
https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8ACAGAAhwEYBuAWACh8jiA6AOThgsqpwGYiAmAgYQIBvKgVFEuJQgFliACgCqCNAQCuSgJQixwygEgxBRWgASAe1gMIYOABUAngAc4BABYA7GAQC8qpXTMWVraOcKwGBpIEUA5uYeGiaABmBLLunl4+RgEwltb2TnQAkgAKAG4ALARYWK4e3plK2bnBBSWlAGyalPGiOj3h0W7eBABExjY2xQD0IwQA1L5oRQAmbtlx8QC+WvFwGFBwO+F9/WKDw2MT07MLACJuUHQA4szZAKIeCHayakuFq9l1P5zDkghtwttuj0cAB2KIxcGQgxHTi0aTcBRKRZdAw6fRiLIg5r5ZxpYa/YGBPIhcERYiEQa0sRJFJkjKGRpEoIkoplSrVWrpBomLnU1plTpHXFSnrnHwU/5rEFM0RInp7A4y3pagYxYb3R4vGDvT7fBUAkFAppgnVq+Jy0bjSYzebw2I62FuxFUTZAA=

But it's just a small thing.  Whatever you prefer is fine.

Reply:
Ok.  I'll make the change.

And I like the https://sharplab.io example!  That is a nice website for this kind of analysis.